src/AST-Core/CompiledMethod.extension.st

```diff
@@ -25,10 +25,9 @@ CompiledMethod >> parseTree [
 	"returns an AST for this method, do not cache it. (see #ast for the cached alternative)"
 	| ast |
 	ast := self methodClass compiler 
-		source: self sourceCode;
 		failBlock: [^ self decompile ];
 		class: self methodClass;
-		parse.
+		parse: self sourceCode.
```

Comment:
Such move of #source: argument is the most of the changes here to remove the state from the compiler